src/vs/editor/browser/services/treeSitter/treeSitterParserService.ts

```diff
@@ -102,7 +105,13 @@ export class TextModelTreeSitter extends Disposable {
 	}
 
 	private async _onDidChangeContent(treeSitterTree: TreeSitterParseResult, changes: IModelContentChange[]) {
-		return treeSitterTree.onDidChangeContent(this.model, changes);
+		const oldTree = treeSitterTree.tree?.copy();
+		await treeSitterTree.onDidChangeContent(this.model, changes);
```

Comment:
Not sure if that is a problem here, but subsequent onDidChangeContent event calls can make multiple promises to get stuck here.
Then line 110 might cause a race condition, depending on which promise resolves first.

Reply:
The actual changes done in `onDidChangecontent` are queued, so don't think we need to worry about the order in which they are resolved: they will always be sequential. 

However, we could end up with `this._onDidChangeParseResult` firing with the same range multiple times.

Reply:
Attempted to address in https://github.com/microsoft/vscode/pull/228986.